packages/node_modules/@ciscospark/react-redux-spark/src/component.js

```diff
@@ -28,73 +28,68 @@ const defaultProps = {
 };
 
 class SparkComponent extends Component {
-  static listenToSparkEvents(spark, props) {
-    spark.listenToAndRun(spark, 'change:canAuthorize', () => {
-      props.updateSparkStatus({authenticated: spark.canAuthorize});
-    });
-
-    spark.listenToAndRun(spark, 'change:isAuthenticating', () => {
-      props.updateSparkStatus({authenticating: spark.isAuthenticating});
-    });
-
-    spark.listenToAndRun(spark.internal.device, 'change:registered', () => {
-      props.updateSparkStatus({registered: spark.internal.device.registered});
-    });
+  constructor(props) {
+    super(props);
 
-    SparkComponent.setupDevice(spark, props);
+    this.storeAndListen = this.storeAndListen.bind(this);
   }
 
   componentDidMount() {
-    const {props} = this;
     const {
       accessToken,
-      guestToken
+      guestToken,
+      spark
     } = this.props;
 
-    const spark = props.spark.get('spark');
+    const sparkInstance = spark.get('spark');
 
-    function storeAndListen(sparkInstance) {
-      props.storeSparkInstance(sparkInstance);
-      SparkComponent.listenToSparkEvents(sparkInstance, props);
-    }
-    if (!spark) {
+    if (!sparkInstance) {
       if (guestToken) {
-        createSparkJwt(guestToken, this.props).then(storeAndListen);
+        createSparkJwt(guestToken, this.props).then(this.storeAndListen);
       }
       else {
-        createSpark(accessToken, this.props).then(storeAndListen);
+        createSpark(accessToken, this.props).then(this.storeAndListen);
       }
     }
     else {
-      SparkComponent.listenToSparkEvents(spark, props);
+      this.listenToSparkEvents(sparkInstance);
     }
   }
 
-  componentWillReceiveProps(nextProps) {
-    const spark = nextProps.spark.get('spark');
-
-    SparkComponent.setupDevice(spark, nextProps);
-  }
-
-  shouldComponentUpdate(nextProps) {
-    const {props} = this;
-
-    return nextProps.spark !== props.spark;
-  }
-
-  static setupDevice(spark, props) {
+  setupDevice(spark) {
```

Comment:
Would be great to add some jsdoc to these methods so we won't be as confused in the future as we were this week 🤣